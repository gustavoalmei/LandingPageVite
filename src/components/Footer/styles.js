import { styled, css } from 'styled-components';
import { Container as TextComponet } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../../components/SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.xsmall};
    }

    & > ${TextComponet} {
      font-size: ${theme.fonts.sizes.xsmall};
    }

    & > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
