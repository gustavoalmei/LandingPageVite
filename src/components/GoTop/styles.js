import { styled, css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;

    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 50%;
  `}
`;
