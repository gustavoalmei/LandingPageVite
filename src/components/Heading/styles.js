import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.large};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1.withConfig({
  // withConfig permite configurar como o Styled Components manipula props específicas.
  shouldForwardProp: (prop) => !['colorDark', 'uppercase'].includes(prop),
  // shouldForwardProp é uma função que decide quais props serão passadas para o DOM. Nesse caso, estamos dizendo que colorDark não deve ser repassado.
})`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
  `}
`;
