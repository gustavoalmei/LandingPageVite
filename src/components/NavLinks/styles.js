import { styled, css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;

    @media (max-width: 768px) {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
