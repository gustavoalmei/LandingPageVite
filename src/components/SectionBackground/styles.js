import { styled, css } from 'styled-components';

const setBackgroundColor = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div.withConfig({
  shouldForwardProp: (prop) => !['background'].includes(prop),
})`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && setBackgroundColor(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
