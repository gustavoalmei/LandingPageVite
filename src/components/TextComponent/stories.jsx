import TextComponent from '.';

export default {
  title: 'TextComponent',
  component: TextComponent,
  args: {
    children: `Lorem, ipsum dolor sit amet consectetur adipisicing elit.
      Expedita, eius? Harum, exercitationem impedit? Debitis voluptatem,
      labore assumenda officiis eum veritatis hic doloribus, nulla mollitia corporis voluptates.
      Maxime sapiente voluptatem asperiores?`,
  },
  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (args) => {
  return (
    <div style={{ maxWidth: '500px' }}>
      <TextComponent {...args} />
    </div>
  );
};
