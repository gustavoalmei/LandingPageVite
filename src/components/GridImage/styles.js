import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    height: 100%;
    object-fit: cover;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.3) rotate(10deg);
    }
  `}
`;
