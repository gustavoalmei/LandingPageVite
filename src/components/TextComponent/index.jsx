import P from 'prop-types';
import * as S from './styles';

export default function TextComponent({ children }) {
  return <S.Container dangerouslySetInnerHTML={{ __html: children }} />;
}

TextComponent.propTypes = {
  children: P.node.isRequired,
};
