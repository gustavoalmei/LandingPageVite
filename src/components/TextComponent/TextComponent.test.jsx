import { screen } from '@testing-library/react';
import { RenderTheme } from '../../styles/handleTheme';
import TextComponent from './index';

describe('<TextComponent/>', () => {
  test('Should be render a text', () => {
    RenderTheme(<TextComponent>Children</TextComponent>);

    expect(screen.getByText('Children')).toBeInTheDocument();
  });

  test('Should be render a text', () => {
    const { container } = RenderTheme(<TextComponent>Children</TextComponent>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
