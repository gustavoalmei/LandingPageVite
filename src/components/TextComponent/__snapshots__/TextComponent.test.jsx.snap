// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextComponent/> Should be render a text 1`] = `
.c0 {
  font-size: 2.5rem;
}

<div
  class="c0"
>
  Children
</div>
`;
